pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1008,6 +1008,20 @@ public void connectionFailed(PulsarClientException exception) {
 
         closeConsumerTasks();
 
+        if (this.deadLetterProducer != null) {
+            try {
+                deadLetterProducer.get().closeAsync();
```

Comment:
Please use asynchronous calls here and make it to be connected to `closeFuture`.